source/MaterialXRuntime/Private/PvtPrim.cpp

```diff
@@ -193,19 +185,19 @@ void PvtPrim::removeAttribute(const RtToken& name)
     }
 }
 
-void PvtPrim::renameAttribute(const RtToken& name, const RtToken& newName)
+
+RtToken PvtPrim::renameAttribute(const RtToken& name, const RtToken& newName)
 {
-    if (getAttribute(newName))
-    {
-        throw ExceptionRuntimeError("An attribute named '" + newName.str() + "' already exists in prim '" + getName().str() + "'");
-    }
+    RtToken uniqueNewName = makeUniqueChildName(newName);
     PvtAttribute* attr = getAttribute(name);
     if (attr)
     {
-        attr->setName(newName);
-        _attrMap[newName] = attr->hnd();
+        attr->setName(uniqueNewName);
+        _attrMap[uniqueNewName] = attr->hnd();
         _attrMap.erase(name);
+        return uniqueNewName;
     }
+    return EMPTY_TOKEN;
```

Comment:
Wasn't sure what to return here, for now am returning the EMPTY_TOKEN, and am checking for it in the application (separate review).

Reply:
I certainly could, the only reason I didn't is that the original code didn't throw in that case.